apps/src/StudioApp.js

```diff
@@ -1323,6 +1323,48 @@ StudioApp.prototype.onMouseMoveVizResizeBar = function (event) {
   this.resizeVisualization(newVizWidth);
 };
 
+StudioApp.prototype.clearVisualizationSize = function () {
+  if ($('#visualizationColumn').hasClass('wireframeShare')) {
+    return;
+  }
+
+  const editorColumn = $(".editor-column");
+  const visualization = document.getElementById('visualization');
+  const visualizationResizeBar = document.getElementById('visualizationResizeBar');
+  const visualizationColumn = document.getElementById('visualizationColumn');
+
+  if (this.isRtl()) {
+    visualizationResizeBar.style.right = '';
+    editorColumn.css('right', '');
+  } else {
+    visualizationResizeBar.style.left = '';
+    editorColumn.css('left', '');
+  }
+
+  visualizationResizeBar.style.lineHeight = '';
+  visualizationColumn.style.maxWidth = '';
+  visualization.style.maxWidth = '';
+  visualization.style.maxHeight = '';
+
+  if (!utils.browserSupportsCssMedia()) {
+    visualization.style.height = '';
+    visualization.style.width = '';
+  }
+
+  applyTransformScaleToChildren(visualization, '');
+
+  $('#soft-buttons').removeClass('soft-buttons-compact');
+
+  const smallFooter = document.querySelector('#page-small-footer .small-footer-base');
+  if (smallFooter) {
+    smallFooter.style.maxWidth = '';
+    const smallPrint = smallFooter.querySelector('small');
+    smallPrint.style.float = '';
+  }
+
+  utils.fireResizeEvent();
```

Comment:
As for the fragility, it may be possible to add a unit test similar to `imageUtilsTest` which takes a thumbnail and compares it against a png file. This seems like it should give consistent results but that would need to be confirmed.

In addition, we can definitely add eyes tests which verify end-to-end that a certain app yields a certain thumbnail in the Projects view within teacher dashboard. This will have the benefit of fuzzy matching.